unified_planning/io/anml_writer.py

```diff
@@ -0,0 +1,330 @@
+
+        for g in self.problem.goals:
+            out.write(f'[ end ] {converter.convert(g)};\n')
+
+        for i, gl in self.problem.timed_goals.items():
+            for g in gl:
+                out.write(f'{self._convert_anml_interval(i)} {converter.convert(g)};\n')
+
+    def print_problem(self):
+        '''Prints to std output the ANML problem.'''
+        self._write_problem(sys.stdout)
+
+    def get_problem(self) -> str:
+        '''Returns the ANML problem.'''
+        out = StringIO()
+        self._write_problem(out)
+        return out.getvalue()
+
+    def write_problem(self, filename: str):
+        '''Dumps to file the ANML problem.'''
+        with open(filename, 'w') as f:
+            self._write_problem(f)
+
+    def _convert_effect(self, effect: 'up.model.Effect',
+                        converter: ConverterToANMLString,
+                        timing: 'up.model.Timing' = None,
+                        spaces_from_left: int = 0) -> str:
+        results: List[str] = []
+        anml_timing = self._convert_anml_timing(timing) if timing is not None else 'start'
+        if effect.is_conditional():
+            results.append(f'when [ {anml_timing} ] {converter.convert(effect.condition)}\n{spaces_from_left*" "}{{')
+        results.append(f'[ {anml_timing} ] ')
+        results.append(converter.convert(effect.fluent))
+        if effect.is_assignment():
+            results.append(' := ')
+        elif effect.is_increase():
+            results.append(' :+= ')
+        elif effect.is_decrease():
+            results.append(' :-= ')
+        else:
+            raise NotImplementedError
+        results.append(f'{converter.convert(effect.value)};\n')
+        if effect.is_conditional():
+            results.append(f'{spaces_from_left*" "}}}\n')
+        return ''.join(results)
+
+    def _convert_anml_timing(self, timing: 'up.model.Timing') -> str:
+        time = 'start' if timing.is_from_start() else 'end'
+        if timing.delay > 0:
+            return f'{time} + {str(timing.delay)}'
+        elif timing.delay == 0:
+            return time
+        else: #timing.delay < 0
+            return f'{time} - {str(timing.delay * (-1))}'
+
+    def _convert_anml_interval(self, interval: 'up.model.TimeInterval') -> str:
+        left_bracket = '(' if interval.is_left_open() else '['
+        right_bracket = ')' if interval.is_left_open() else ']'
+        return f'{left_bracket} {self._convert_anml_timing(interval.lower)}, {self._convert_anml_timing(interval.upper)} {right_bracket}'
+
+def _is_valid_anml_name(name: str) -> bool:
+    regex = re.compile('^[a-zA-Z]+.*')
+    if re.match(regex, name) is None or name in ANML_KEYWORDS: # If the name does not start with an alphabetic char or is a keyword
+        return False
+    return name.isidentifier() #NOTE Here I am creating a dependency between python identifiers and ANML names. For now they are (almost) the same, but in future versions?
```

Comment:
If you implement the strict suggestion above, you can safely return `True` as the regexp I suggest works for both languages